src/rgw/rgw_common.h

```diff
@@ -1693,6 +1693,7 @@ struct req_state {
   bool enable_usage_log;
   uint8_t defer_to_bucket_acls;
   uint32_t perm_mask;
+  string subuser_name;
```

Comment:
I'm not convinced that we need to extend  `req_state` in this particular case. IIRC there was an idea to cut down `perm_mask` after moving all authenticators to the new auth infrastructure.
An alternative would be to extend the `RGWIdentityApplier` with something like:
```cpp
    virtual boost::optional<boost::string_ref> get_subuser_name() const
```
I'm just thinking loudly and definitely stay open for discussion.